astropy/cosmology/parameter/_descriptors.py

```diff
@@ -19,20 +19,52 @@
 
 @dataclass(frozen=True, **_dataclass_kwargs)
 class ParametersAttribute:
-    """Immutable mapping of the Parameters.
+    """Immutable mapping of the :class:`~astropy.cosmology.Parameter` objects or values.
 
-    If accessed from the class, this returns a mapping of the Parameter
-    objects themselves.  If accessed from an instance, this returns a
-    mapping of the values of the Parameters.
+    If accessed from the :class:`~astropy.cosmology.Cosmology` class, this returns a
+    mapping of the :class:`~astropy.cosmology.Parameter` objects themselves.  If
+    accessed from an instance, this returns a mapping of the values of the Parameters.
+
+    This class is used to implement the ``parameters`` attribute of
+    :class:`~astropy.cosmology.Cosmology`.
 
     Parameters
     ----------
     attr_name : str
         The name of the class attribute containing the Parameter objects.
+
+    Examples
+    --------
+    This class works as a descriptor on the containing class. When accessed from the
+    class, it returns the attribute specified by ``attr_name``. When accessed from an
+    instance, it returns a proxy mapping of the value for each key in ``attr_name``.
+
+    In the following example ``attr_name`` is set to ``_attr_map``, which is not a
+    dictionary, but a tuple of strings. When accessed from the class, the tuple is
+    returned. When accessed from an instance, a mapping of the values of the attributes
+    named in the tuple is returned.
+
+    >>> class Obj:
+    ...     a = 1
+    ...     b = 2
+    ...     c = 3
+    ...     _attr_map = ("a", "b", "c")
+    ...     attr = ParametersAttribute(attr_name="_attr_map")
+
+    >>> Obj.attr
+    ('a', 'b', 'c')
+
+    >>> obj = Obj()
+    >>> obj.attr
+    mappingproxy({'a': 1, 'b': 2, 'c': 3})
     """
 
-    attr_name: str | None = None
-    """The name of the class attribute containing the Parameter objects."""
+    attr_name: str
```

Comment:
Better as plural? `attr_names`? Or even `parameter_names`?

Reply:
It's a singular attribute on `Cosmology`. Maybe `class_attr_name`?

Reply:
It is a singular attribute, but doesn't it contain more than one name of attributes/descriptors/parameters inside? Or have I just confused myself?

Reply:
The `ParametersAttribute`` uses `attr_name="_parameters"` to find the `Cosmology._parameters: MappingProxyType[str, Parameter]` class attribute.  It's `getattr(Cosmology, attr_name)`.